moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -590,55 +590,32 @@ void RobotState::updateLinkTransforms()
 
 void RobotState::updateLinkTransformsInternal(const JointModel* start)
 {
-  const std::vector<const LinkModel*>& links = start->getDescendantLinkModels();
-  if (!links.empty())
-  {
-    const LinkModel* parent = links[0]->getParentLinkModel();
+  for(const LinkModel* link : start->getDescendantLinkModels()) {
+    const LinkModel* parent = link->getParentLinkModel();
     if (parent)
     {
-      if (links[0]->parentJointIsFixed())
-        global_link_transforms_[links[0]->getLinkIndex()].matrix().noalias() =
-            global_link_transforms_[parent->getLinkIndex()].matrix() * links[0]->getJointOriginTransform().matrix();
+      if (link->parentJointIsFixed())
```

Comment:
could you please add a comment to the `if(parent)` line to explain that only the root link of `start` might not have a parent?